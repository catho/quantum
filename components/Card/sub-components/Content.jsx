import styled from 'styled-components';
import PropTypes from 'prop-types';

import {
  spacing,
  baseFontSize as defaultBaseFontSize,
} from '../../shared/theme';

const Content = styled.div`
  margin: 0;

  ${({
    theme: {
      baseFontSize,
      spacing: { small, medium },
    },
  }) => `
    padding: ${small}px ${medium}px;
    font-size: ${baseFontSize * 0.875}px;
  `}
`;

Content.propTypes = {
  theme: PropTypes.shape({
    baseFontSize: PropTypes.number,
    spacing: PropTypes.object,
  }),
};

Content.defaultProps = {
  theme: {
    spacing,
    baseFontSize: defaultBaseFontSize,
  },
};

Content.displayName = 'Card.Content';

export default Content;
