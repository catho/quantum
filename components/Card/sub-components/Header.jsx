import styled from 'styled-components';
import PropTypes from 'prop-types';

import { spacing } from '../../shared/theme';

const Header = styled.header`
  display: flex;

  ${({
    theme: {
      spacing: { medium },
    },
  }) => `
    padding: ${medium}px ${medium}px 0;

    > * {
      margin-right: ${medium}px;
    }

    > *:last-child {
      margin-right: 0;
    }
  `}
`;

Header.propTypes = {
  theme: PropTypes.shape({
    spacing: PropTypes.object,
  }),
};

Header.defaultProps = {
  theme: {
    spacing,
  },
};

Header.displayName = 'Card.Header';

export default Header;
