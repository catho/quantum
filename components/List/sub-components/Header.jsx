import PropTypes from 'prop-types';
import styled from 'styled-components';
import { spacing } from '../../shared/theme';

const Title = styled.span`
  ${({
    theme: {
      spacing: { xxsmall },
    },
  }) => `
  margin: ${xxsmall}px 0;
`}
`;

Title.displayName = 'Header';

const Header = ({
  children,
  theme = {
    spacing,
  },
}) => <Title theme={theme}>{children}</Title>;

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  theme: PropTypes.shape({
    spacing: PropTypes.object,
  }),
};

Header.displayName = 'List.Header';

export default Header;
