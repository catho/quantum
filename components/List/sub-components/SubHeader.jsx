import PropTypes from 'prop-types';
import styled from 'styled-components';
import { baseFontSize } from '../../shared/theme';

const Description = styled.span`
  ${{
    theme: {
      baseFontSize,
    },
  }}
  font-size: ${baseFontSize * 0.75}px;
`;

Description.displayName = 'SubHeader';

const SubHeader = ({ children, theme }) => (
  <Description theme={theme}>{children}</Description>
);

SubHeader.defaultProps = {
  theme: {
    baseFontSize,
  },
};

SubHeader.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  theme: PropTypes.shape({
    baseFontSize: PropTypes.number,
  }),
};

SubHeader.displayName = 'List.SubHeader';

export default SubHeader;
