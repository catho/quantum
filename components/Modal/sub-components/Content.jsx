import styled from 'styled-components';
import PropTypes from 'prop-types';
import Card from '../../Card';
import { spacing } from '../../shared/theme';

const Content = styled(Card.Content)`
  font-size: 16px;
  max-height: 70vh;
  overflow-y: auto;
  padding: ${({
    theme: {
      spacing: { medium },
    },
  }) => `${medium}px`};
`;

Content.propTypes = {
  theme: PropTypes.shape({
    spacing: PropTypes.object,
  }),
};

Content.defaultProps = {
  theme: { spacing },
};

export default Content;
