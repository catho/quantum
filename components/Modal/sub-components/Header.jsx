import styled from 'styled-components';
import PropTypes from 'prop-types';
import Card from '../../Card';
import { spacing } from '../../shared/theme';

const Header = styled(Card.Header)`
  padding: ${({
    theme: {
      spacing: { medium },
    },
  }) => `${medium}px ${medium}px 0`};
`;

Header.propTypes = {
  theme: PropTypes.shape({
    spacing: PropTypes.object,
  }),
};

Header.defaultProps = {
  theme: { spacing },
};

export default Header;
