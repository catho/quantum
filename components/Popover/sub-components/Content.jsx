import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { components, spacing, colors, breakpoints } from '../../shared/theme';
import getArrow from '../arrowProperties';
import Button from '../../Button';
import { shadow } from '../../shared';

const _getColors = ({
  skin,
  theme: {
    components: {
      popover: {
        skins: {
          [skin]: { text, background },
        },
      },
    },
  },
  inverted,
}) => `
  background-color: ${inverted ? text : background}
  color: ${inverted ? background : text}
`;

const _getTextColor = ({
  skin,
  theme: {
    components: {
      popover: {
        skins: {
          [skin]: { background, text },
        },
      },
    },
  },
  inverted,
}) => `color: ${inverted ? background : text};`;

const PopoverContent = styled.div`
  ${shadow(5)};
  align-items: start;
  display: flex;
  border-radius: 4px;
  font-size: 16px;
  ${({
    theme: {
      spacing: { xsmall },
    },
  }) => `padding: ${xsmall}px;`}

  position: absolute;
  line-height: 0;
  transition: opacity 0.2s ease-in-out, visibility 0.2s ease-in-out;
  z-index: 100;

  &:before {
    ${getArrow};
  }

  ${_getColors}
`;

const CloseButton = styled(Button.Icon).attrs({
  icon: 'close',
})`
  ${_getTextColor}
  display: inherit;
  height: auto;
  ${({
    theme: {
      spacing: { medium },
    },
  }) => `margin-left: ${medium}px;`}

  opacity: 0.8;
  padding: 0;
  transition: opacity 0.4s ease;
  width: auto;

  :hover {
    background: none;
    opacity: 1;
    ${_getTextColor}
  }
`;

const PopoverChildren = styled.div``;

class Content extends Component {
  constructor(props) {
    super(props);

    this.wrapper = document.createElement('section');
  }

  componentDidMount() {
    document.body.appendChild(this.wrapper);
  }

  componentWillUnmount() {
    document.body.removeChild(this.wrapper);
  }

  render() {
    const {
      placement,
      children,
      onPopoverClose,
      theme,
      skin,
      inverted,
      ...rest
    } = this.props;

    return ReactDOM.createPortal(
      <PopoverContent
        theme={theme}
        inverted={inverted}
        placement={placement}
        skin={skin}
        ref={element => {
          this.innerContentRef = element;
        }}
        {...rest}
      >
        <PopoverChildren>{children}</PopoverChildren>
        <CloseButton
          skin={skin}
          theme={theme}
          inverted={inverted}
          onClick={onPopoverClose}
        />
      </PopoverContent>,
      this.wrapper,
    );
  }
}

CloseButton.displayName = 'CloseButton';
PopoverChildren.displayName = 'PopoverChildren';

Content.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  inverted: PropTypes.bool,
  placement: PropTypes.oneOf(['top', 'right', 'bottom', 'left']),
  onPopoverClose: PropTypes.func,
  skin: PropTypes.oneOf(['neutral', 'success', 'warning', 'error']),
  theme: PropTypes.shape({
    colors: PropTypes.object,
    spacing: PropTypes.object,
    breakpoints: PropTypes.array,
    components: PropTypes.shape({
      popover: PropTypes.object,
      button: PropTypes.object,
    }),
  }),
};

Content.defaultProps = {
  inverted: false,
  placement: 'top',
  onPopoverClose: () => {},
  skin: 'neutral',
  theme: {
    colors,
    spacing,
    breakpoints,
    components: {
      popover: components.popover,
      button: components.button,
    },
  },
};

export default Content;
