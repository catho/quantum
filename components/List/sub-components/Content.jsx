import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Header from './Header';
import SubHeader from './SubHeader';

const ContentBox = styled.div`
  display: flex;
  flex-direction: column;
`;

ContentBox.displayName = 'Content';

class Content extends React.Component {
  _renderContent = content => {
    const { header: propsHeader, subheader: propsSubHeader } = this.props;

    if (typeof content === 'string') {
      return <Header>{content}</Header>;
    }

    const { header = propsHeader, subheader = propsSubHeader } = content;

    return (
      <>
        <Header>{header}</Header>

        {subheader && <SubHeader>{subheader}</SubHeader>}
      </>
    );
  };

  render() {
    const { content, children } = this.props;

    return <ContentBox>{children || this._renderContent(content)}</ContentBox>;
  }
}

Content.defaultProps = {
  subheader: '',
  header: '',
  content: {},
  children: null,
};

Content.propTypes = {
  header: PropTypes.string,
  subheader: PropTypes.string,
  content: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({
      header: PropTypes.string,
      subheader: PropTypes.string,
    }),
  ]),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

Content.displayName = 'List.Content';

export default Content;
