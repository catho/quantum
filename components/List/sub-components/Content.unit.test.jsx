import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Content from './Content';
import Header from './Header';
import SubHeader from './SubHeader';

const content = 'content';
const header = 'header';
const subheader = 'subheader';

describe('<Content />', () => {
  it('Should match the snapshot', () => {
    const wrapper = mount(<Content />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render Header component', () => {
    const wrapper = mount(<Content content={content} />);

    expect(wrapper.render().text()).toBe(content);
    expect(wrapper.find(Header)).toHaveLength(1);
  });

  it('accept any children as a content', () => {
    const wrapper = mount(<Content>{content}</Content>);

    expect(wrapper.render().text()).toBe(content);
    expect(wrapper.find(Header)).toHaveLength(0);
  });

  it('render Header and SubHeader components', () => {
    const contentWrapper = mount(<Content content={{ header, subheader }} />);
    const propWrapper = mount(
      <Content header={header} subheader={subheader} />,
    );
    // ###
    // ###
    // ###
    // ### FAZER O HEADER E SUBHEADER FUNCIONAREM QDO PASSADO VIA PROPS
    // console.log('###### PROP', propWrapper.find(Header)).toHaveLength(2);

    expect(contentWrapper.find(Header)).toHaveLength(1);
    expect(contentWrapper.render().text()).toContain(header);
    expect(contentWrapper.find(SubHeader)).toHaveLength(1);
    expect(contentWrapper.render().text()).toContain(subheader);

    expect(propWrapper.find(Header)).toHaveLength(1);
    expect(propWrapper.render().text()).toContain(header);
    expect(propWrapper.find(SubHeader)).toHaveLength(1);
    expect(propWrapper.render().text()).toContain(subheader);
  });
});
