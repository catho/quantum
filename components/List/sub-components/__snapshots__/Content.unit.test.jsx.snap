// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> Should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List.Content
    content={Object {}}
    header=""
    subheader=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <React.Fragment>
        <List.Header>
          
        </List.Header>
        
      </React.Fragment>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <List.Header>
            
          </List.Header>,
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
        "",
      ],
      "type": Symbol(react.fragment),
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Content__ContentBox-sc-1fmj11p-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;flex-direction:column;",
        ],
      },
      "displayName": "Content",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Content__ContentBox-sc-1fmj11p-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <React.Fragment>
          <List.Header>
            
          </List.Header>
          
        </React.Fragment>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <List.Header>
              
            </List.Header>,
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
          "",
        ],
        "type": Symbol(react.fragment),
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Content__ContentBox-sc-1fmj11p-0",
          "isStatic": true,
          "rules": Array [
            "display:flex;flex-direction:column;",
          ],
        },
        "displayName": "Content",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Content__ContentBox-sc-1fmj11p-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
